packages/formspree-react/src/useForm.ts

```diff
@@ -105,10 +105,11 @@ const useForm = (
     if (typeof extraData === 'object') {
       for (const prop in extraData) {
         if (typeof extraData[prop] === 'function') {
-          let extraDataValue = (
-            extraData[prop] as Exclude<ExtraDataValue, string>
-          ).call(null);
-          if (extraDataValue instanceof Promise) {
+          let extraDataValue = (extraData[prop] as Exclude<
+            ExtraDataValue,
+            string
+          >).call(null);
+          if (typeof extraDataValue?.then === 'function') {
```

Comment:
What was the issue with this not being detected as a promise?

Reply:
The `react-google-recaptcha-v3` lib doesn't return a Promise instance, it instead returns a `thenable` object (basically an object with a then function). So checking instanceof doesn't work. Also generally checking instanceof on promises appears to be bad practice since there are lots of promise implementation libraries that don't actually return promise instances. This is tolerated by javascript, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/await.